### PR TITLE
fix(benchmarks,runtime-interface): clear store by counter when runnining benchmarks

### PR DESCRIPTION
**Pull request by ByteNacked**
Resolves # .

Fixes the issue of a 10x slowdown in `onetime syscall` benchmarks.

[Original problem](https://github.com/gear-tech/gear/pull/4136/commits/58643b209e890d64945d2f466808745fd52b49f2)

@reviewer-or-team

### CODE REVIEW

**Comment by ByteNacked** [comment]
TODO: update weights for onetime syscall benches

File: runtime-interface/sandbox/Cargo.toml [hunk 1]
```diff
@@ -28,3 +28,4 @@ std = [
   "sp-runtime-interface/std",
   "sp-wasm-interface/std",
 ]
+runtime-benchmarks = []
```
Comment (breathx) [inline runtime-interface/sandbox/Cargo.toml:31]: Newline

File: runtime-interface/Cargo.toml [hunk 2]
```diff
@@ -53,3 +53,6 @@ std = [
 	"ark-scale/std",
 	"sha2/std"
 ]
+runtime-benchmarks = [
+	"gear-sandbox-interface/runtime-benchmarks"
+]
```
Comment (breathx) [inline runtime-interface/Cargo.toml:58]: Newline

**Review by breathx — APPROVED** [review]
Doesn't this affect other benches, not only onetime syscalls?

File: runtime-interface/Cargo.toml (hunk 2 above)
Comment (ByteNacked) [reply runtime-interface/Cargo.toml]: Fixed

File: runtime-interface/sandbox/Cargo.toml (hunk 1 above)
Comment (ByteNacked) [reply runtime-interface/sandbox/Cargo.toml]: Fixed

**Comment by ByteNacked** [comment]
> Doesn't this affect other benches, not only onetime syscalls?

It's not affect normal benchmarks because they run with :

```
BENCHMARK_STEPS=50
BENCHMARK_REPEAT=20
```

and store cleared normally (by method `Sanboxes::get` or by creating new thread).

**Comment by gshep** [comment]
Another option is to use `Vec::with_capacity` with some good value for collections of `SandboxComponents`. It would allow to avoid redundant reallocations which are the reason of regression.

**Comment by ByteNacked** [comment]
> Another option is to use `Vec::with_capacity` with some good value for collections of `SandboxComponents`. It would allow to avoid redundant reallocations which are the reason of regression.

Will check, but I think the reason is not in big sizes of these `Vecs` (Instance, memory suppose to be just ptr size) but due to internal indexes inside wasmer store object (I think we not supposed to store >= 1000 memory and instances in it).

**Comment by ByteNacked** [comment]
> Another option is to use `Vec::with_capacity` with some good value for collections of `SandboxComponents`. It would allow to avoid redundant reallocations which are the reason of regression.

Checked https://github.com/gear-tech/gear/actions/runs/10576870738/job/29303638506 , it seems doesn't work

**Comment by ByteNacked** [comment]
Updated onetime syscalls' weights

**Pull request merged**